src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -129,59 +128,78 @@ private void AnalyzeTemplate(FileInfo templateFilePath, FileInfo parametersFileP
                 }
 
                 Console.WriteLine($"{IndentedNewLine}Rules passed: {passedEvaluations}");
+
+                return 1;
             }
             catch (Exception exp)
             {
                 Console.WriteLine($"An exception occured: {GetAllExceptionMessages(exp)}");
+                return -1;
             }
         }
 
         private void AnalyzeDirectory(DirectoryInfo directoryPath)
         {
             try {
+
+                if (!directoryPath.Exists)
+                {
+                    Console.WriteLine($"Invalid directory ({directoryPath})");
+                    return;
+                }
+
                 // Find files to analyze
                 List<FileInfo> filesToAnalyze = new List<FileInfo>();
-                FindTemplateInDirectory(directoryPath, filesToAnalyze);
+                FindJsonFilesInDirectoryRecursive(directoryPath, filesToAnalyze);
 
                 // Log root directory
                 string directoryMetadata = Environment.NewLine + Environment.NewLine + $"Directory: {directoryPath}";
                 Console.WriteLine(directoryMetadata);
 
+                int numOfSuccesses = 0;
+                int numOfFails = 0;
                 foreach (FileInfo fileToAnalyze in filesToAnalyze)
                 {
-                    AnalyzeTemplate(fileToAnalyze, null, false);
+                    int res = AnalyzeTemplate(fileToAnalyze, null, false);
+                    if (res == 1) 
+                    {
+                        numOfSuccesses++;
+                    }
+                    else if (res == -1) 
+                    {
+                        numOfFails++;
+                    }
+                }
+
+                Console.WriteLine(Environment.NewLine + $"Analyzed {numOfSuccesses} files.");
+                if (numOfFails > 0) {
+                    Console.WriteLine($"Unable to analyze {numOfFails} files.");
```

Comment:
Minor: Update to `file(s)`. Also on line 174.

Reply:
It would be good to at least print out the file paths that failed.  Some message with the failure reason would be good too, but at least the file that failed is a good start for now.